src/adapter/src/explain/optimizer_trace.rs

```diff
@@ -260,18 +258,40 @@ impl OptimizerTrace {
         for<'a> Explainable<'a, T>: Explain<'a, Context = ExplainContext<'a>>,
     {
         if let Some(trace) = self.0.downcast_ref::<PlanTrace<T>>() {
+            // Get a handle of the associated `PlanTrace<UsedIndexes>`.
+            let used_indexes_trace = self.0.downcast_ref::<PlanTrace<UsedIndexes>>();
+
             trace
                 .drain_as_vec()
                 .into_iter()
                 .map(|mut entry| {
-                    // update the context with the current time
+                    // Update the context with the current time.
                     context.duration = entry.full_duration;
+
+                    // Try to find the UsedIndexes instance for this entry.
+                    let used_indexes = used_indexes_trace.map(|t| t.used_indexes_for(&entry.path));
```

Comment:
Lookup a `used_indexes: UsedIndexes` for the current path from the `used_indexes_trace`.

See `PlanTrace<UsedIndexes>::used_indexes_for`.